tests/foreman/api/test_discoveredhost.py

```diff
@@ -55,9 +61,94 @@ def _create_discovered_host(name=None, ipaddress=None, macaddress=None):
     })
 
 
+class HostNotDiscoveredException(Exception):
+    """Raised when host is not discovered"""
+
+
 class DiscoveryTestCase(APITestCase):
     """Implements tests for foreman discovery feature"""
 
+    def _assertdiscoveredhost(self, hostname):
+        """Check if host is discovered and information about it can be
+        retrieved back
+
+        Introduced a delay of 300secs by polling every 10 secs to get expected
+        host
+        """
+        for _ in range(30):
+            discovered_host = entities.DiscoveredHost().search(
+                query={'search': 'name={}'.format(hostname)})
+            if discovered_host:
+                break
+            time.sleep(10)
+        else:
+            raise HostNotDiscoveredException(
+                "The host {} is not discovered within 300 seconds".format(hostname))
+        return discovered_host[0]
+
+    @classmethod
+    @skip_if_not_set('vlan_networking')
+    def setUpClass(cls):
+        """Steps to Configure foreman discovery
+
+        1. Build PXE default template
+        2. Create Organization/Location
+        3. Update Global parameters to set default org and location for
+           discovered hosts.
+        4. Enable auto_provision flag to perform discovery via discovery
+           rules.
+        """
+        super(DiscoveryTestCase, cls).setUpClass()
+
+        # Build PXE default template to get default PXE file
+        entities.ConfigTemplate().build_pxe_default()
+        # let's just modify the timeouts to speed things up
+        ssh.command("sed -ie 's/TIMEOUT [[:digit:]]\\+/TIMEOUT 1/g' "
+                    "/var/lib/tftpboot/pxelinux.cfg/default")
+        ssh.command("sed -ie '/APPEND initrd/s/$/ fdi.countdown=1/' "
+                    "/var/lib/tftpboot/pxelinux.cfg/default")
```

Comment:
Is such modification of host files safe? Isn't there a risk that something else will run at the same time or later, but will require different values?

Reply:
@mirekdlugosz fair point. However, I'd consider `fdi.countdown=1` a default for our automation. It'd be probably better to set this timeout on a global level as a part of the provisioning job.
Anyway, the risk is not that big (we have no aspirations on testing different pxelinux configs for now)

Reply:
Yes, Its less likely that other tests are using this setting than Discovery and even if its provisioning then its 1 as @rplevka said.

@rplevka We won't do that at the global level as we will end up two different component tests dependent on single settings and hence change in either one component in one discovery/provisioning could be the cause for another to fail.

So let's handle those settings independently and as per the convinience of components.